numpy/lib/nanfunctions.py

```diff
@@ -1008,10 +1012,12 @@ def _nanmedian_small(a, axis=None, out=None, overwrite_input=False):
     for i in range(np.count_nonzero(m.mask.ravel())):
         warnings.warn("All-NaN slice encountered", RuntimeWarning,
                       stacklevel=4)
+
+    fill_value = np.timedelta64("NaT") if m.dtype.kind == "m" else np.nan
```

Comment:
Just noting here that datetime64 also has NaT.

Reply:
`datetime64` is not supported by `np.median` though, so I think skipping it should be fine.

Reply:
> datetime64 is not supported by np.median

Seems to work:
```
In [1]: a = np.full((3,3), np.nan).astype("M8[ms]")                             

In [2]: nanmedian(a)                                                            
/home/charris/.local/lib/python3.9/site-packages/numpy-1.21.0.dev0+1752.gef6595517f-py3.9-linux-x86_64.egg/numpy/lib/nanfunctions.py:1113: RuntimeWarning: All-NaN slice encountered
  r, k = function_base._ureduce(a, func=_nanmedian, axis=axis, out=out,
Out[2]: nan

In [3]: a                                                                       
Out[3]: 
array([['NaT', 'NaT', 'NaT'],
       ['NaT', 'NaT', 'NaT'],
       ['NaT', 'NaT', 'NaT']], dtype='datetime64[ms]')
```

Reply:
I wonder what the argument is for median not supporting `datetime64`? I see that `mean` fails because it uses addition, but I think a mean date still makes sense.

Reply:
> Seems to work for `nanmedian` though.

Interesting, it seems by passing an all-nan array triggers a code-path that _does_ handle `datetime64`.
The moment nan is replaced by an actual date an exception is triggerd though.
``` python
In [81]: np.nanmedian(np.datetime64(1))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
...
UFuncTypeError: ufunc 'add' cannot use operands with types dtype('<M8[Y]') and dtype('<M8[Y]')
```

Reply:
> I wonder what the argument is for median not supporting `datetime64`?

Honestly? I suspect that this was just an oversight from when the function was originally implemented.